hrishioa/lumentis
# Lumentis Goes Bilingual and maybe Multilingual: Adding Korean Support and i18n Framework

Annyeonghaseyo, Lumentis! 🇰🇷

This PR adds internationalization (i18n) to Lumentis, adding support for Korean language alongside English! 

### Summary of Changes 

1. Implemented i18next for internationalization:

-  i18next configuration in app.ts

2. Updated user-facing strings:

- Replaced hardcoded English strings with calls to the translation function t()
- This way, all prompts and messages are mapped to the translation function

3. Created translation files:

- Added src/locales/en/translation.json for English
- Added src/locales/ko/translation.json for Korean


4. Modified app.ts to support language selection:

- Added a language selection prompt at the start of the wizard
- Implemented language switching using i18next


Overall:  

- Users can now choose between English and Korean at the start of the wizard
- All subsequent prompts and messages will be displayed in the chosen language
- The change is backwards compatible; English remains the default language
- No changes to the core functionality of Lumentis; only the user interface is affected

### Testing 

1. Tested: 

- Tested the wizard flow in both English and Korean
- Verified that all major translated strings are displayed correctly
- Ensured that the language selection does not interfere with the existing workflow

2. To test this PR:

- Install the new dependencies:` npm install i18next i18next-fs-backend`
- Run lumentis 
- Select a language when prompted: English or 한국어 (Korean)
- Proceed through the wizard, verifying that all text is in the chosen language

### Further Language Support 
For other languages:
1. Created the translated script following the format of src/locales/en/translation.json 
2. Create a folder under src/locales/[abbreviated language code]
3. Adjust the app.ts accordingly 

### What Still Needs to be Added

1. Korean font: Right now, Korean words are displayed a bit unnaturally because there's a lot more gap between character. Seeing if this can be improved by adding a font directly to the repo. 
2. Not all messages are in Korean: Some messages that are not coming from app.ts  are not translated to Korean.
- Example: `(Press <space> to select, <a> to toggle all, <i> to invert selection, and <enter> to proceed)`
3. Adjust the structure of app.ts to easily support adding additional languages
